### PR TITLE
integration: cancel operation for unsynced watcher

### PR DESCRIPTION
**Pull request by gyuho**
Related https://github.com/coreos/etcd/issues/4216.

### CODE REVIEW

File: integration/v3_grpc_test.go [hunk 1]
```diff
@@ -428,6 +428,11 @@ func TestV3WatchFromCurrentRevision(t *testing.T) {
 
 // TestV3WatchCancel tests Watch APIs cancellation.
 func TestV3WatchCancel(t *testing.T) {
+	testV3WatchCancel(t, 0) // synced
+	testV3WatchCancel(t, 1) // unsynced
```
Comment (heyitsanthony) [inline integration/v3_grpc_test.go]: I think it'd be easier to debug test failures if this is split into `TestV3WatchCancelSynced` and `TestV3WatchCancelUnsynced`

**Comment by gyuho** [comment]
@heyitsanthony @xiang90 Please review. I am still investigating https://github.com/coreos/etcd/pull/4242 on the side, where unsynced watchers are blocking in CI environments. This one works though.

Adding more tests to unsynced cases should help.

Thanks,

File: integration/v3_grpc_test.go (hunk 1 above)
Comment (gyuho) [inline integration/v3_grpc_test.go]: Ok will split. THanks,

**Comment by heyitsanthony** [comment]
lgtm after splitting

**Comment by gyuho** [comment]
Thanks, just fixed. Will merge after CI pases.

**Pull request merged**